File: app/src/main/java/it/cammino/risuscito/objects/PosizioneItem.java
package it.cammino.risuscito.objects;

/**
 * Created by marcello.battain on 08/04/2015.
 */
public class PosizioneItem {

    private int pagina;
    private String titolo;
    private String colore;
    private int idCanto;
    private String timestamp;
    private String source;
    private boolean mSelected;

    public PosizioneItem(int pagina, String titolo, String colore, int idCanto, String source, String timestamp) {
        this.setTitolo(titolo);
        this.setPagina(pagina);
        this.setColore(colore);
        this.setIdCanto(idCanto);
        this.setSource(source);
        this.setTimestamp(timestamp);
        this.setmSelected(false);
    }

    public int getPagina() {
        return pagina;
    }

    public void setPagina(int pagina) {
        this.pagina = pagina;
    }

    public String getTitolo() {
        return titolo;
    }

    public void setTitolo(String titolo) {
        this.titolo = titolo;
    }

    public String getColore() {
        return colore;
    }

    public void setColore(String colore) {
        this.colore = colore;
    }

    public int getIdCanto() {
        return idCanto;
    }

    public void setIdCanto(int idCanto) {
        this.idCanto = idCanto;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public boolean ismSelected() {
        return mSelected;
    }

    public void setmSelected(boolean mSelected) {
        this.mSelected = mSelected;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }

}

File: app/src/main/java/it/cammino/risuscito/objects/PosizioneItem.kt
package it.cammino.risuscito.objects

import android.graphics.Color
import androidx.annotation.StringRes
import com.mikepenz.fastadapter.ui.utils.StringHolder
import it.cammino.risuscito.utils.Utility.helperSetColor

fun posizioneItem(block: PosizioneItem.() -> Unit): PosizioneItem = PosizioneItem().apply(block)

@Suppress("unused")
class PosizioneItem {

    var color: Int = Color.WHITE
    var title: StringHolder? = null
        private set
    var page: StringHolder? = null
        private set
    var source: StringHolder? = null
        private set
    var idCanto: Int = 0
    var timestamp: StringHolder? = null
    private var mSelected: Boolean = false

    fun ismSelected(): Boolean {
        return mSelected
    }

    fun setmSelected(mSelected: Boolean) {
        this.mSelected = mSelected
    }

    fun withTitle(title: String): PosizioneItem {
        this.title = StringHolder(title)
        return this
    }

    fun withTitle(@StringRes titleRes: Int): PosizioneItem {
        this.title = StringHolder(titleRes)
        return this
    }

    fun withPage(page: String): PosizioneItem {
        this.page = StringHolder(page)
        return this
    }

    fun withPage(@StringRes pageRes: Int): PosizioneItem {
        this.page = StringHolder(pageRes)
        return this
    }

    fun withSource(src: String): PosizioneItem {
        this.source = StringHolder(src)
        return this
    }

    fun withSource(@StringRes srcRes: Int): PosizioneItem {
        this.source = StringHolder(srcRes)
        return this
    }

    fun withColor(color: String): PosizioneItem {
        this.color = helperSetColor(color)
        return this
    }

    fun withTimestamp(timestamp: String): PosizioneItem {
        this.timestamp = StringHolder(timestamp)
        return this
    }

    fun withId(id: Int): PosizioneItem {
        this.idCanto = id
        return this
    }

}
